sudospawner/spawner.py

```diff
@@ -43,9 +43,15 @@ def do(self, action, **kwargs):
         yield p.stdin.write(json.dumps(kwargs).encode('utf8'))
         p.stdin.close()
         data = yield p.stdout.read_until_close()
+        # Read again if output is not json
+        if not data.decode('utf8').startswith('{'):
+            data = yield p.stdout.read_until_close()
```

Comment:
I wonder if it's getting both stderr and stdout, triggering two close events. I think there might be another way.